Added a list of skills an RSE might possess. (#67)
vsoch/usrse.github.io: pages/resources/skills.md

---
layout: page

> title: RSE Resources

Reviewer: ```suggestion
title: RSE Skills
```

permalink: /resources/skills/

> subtitle: Resources for RSEs

Reviewer: ```suggestion
subtitle: Skillsets used by RSEs
```

---

> The following is a list of skills that an RSE might possess. An RSE typically has experience with a subset of these skills. By no means is this list meant to be understood as what an RSE should know but rather as a list that can be used to create a profile for an RSE or an RSE position by picking relevant skills. This list is also not comprehensive. If you find something is missing, please contact the Training and Education Working Group.

Reviewer: ```suggestion
The following is a list of skills that an RSE might possess. The US-RSE Education and Training Group maintains and publishes this list in an effort to better define RSE roles and to help direct individuals looking to invest in learning new skills. An RSE typically has experience with a subset of these skills. This list is not intended to define what an RSE should know, but rather as a list that can be used to create a profile or job description by choosing relevant skills. This list is also not comprehensive. If you have additions, please contact the Training and Education Working Group.
```

Reviewer: I think we might need a little more justification for why we are posting this, and what our intention is for it. I tried to add a bit to help.

> **Software Development**

Reviewer 2: ```suggestion
## Software Development
```
...so that it should up as a heading with an anchor.

  - Requirements gathering/analysis/formulation
  - Technology evaluation
  - Packaging/Releasing
  - Programming/implementation
    - Software Architecture
    - Cross-platform development
    - Maintenance
        - Bug fixing
        - Version upgrades
  - Testing
  - Distributed systems
    - Cloud computing
    - Grid computing
  - Client-server architectures/web applications
    - Frontend
        - Frameworks

>         - UX

Reviewer 2: ```suggestion
        - User experience (UX)
```

        - Testing
    - Backend
    - Web services (API development)
  - Desktop applications
  - Embedded systems/software
  - Real-time systems software

**User interface**
  - UX design
  - Graphic design

**Areas of CS**
  - Algorithm
     - Development
     - Evaluation
  - Framework development
  - Formal languages and automata theory
  - Logic

>   - Randomness, geometry and discrete structures

Reviewer 3: Seems odd to me to put randomness together with geometry and discrete structures.

  - Semantics and reasoning
  - Agent-based modelling
  - Artificial intelligence
     - Machine Learning (Deep learning, neural nets, etc.)
  - Computer vision
  - Information retrieval
  - Data mining
  - High performance computing
  - Signal detection/processing
  - Cryptography
      - Blockchain
  - Compiler construction/design
  - GPU programming
  - Scientific computing
  - Computer networking
  - Operating systems

**Domain-specific topics**

>   - GIS

Reviewer 2: ```suggestion
  - Geographic Information System (GIS)
```

  - Bioinformatics
  - Industrial control systems
  - Simulation

**Around Software**
  - Project management
  - Software aspects
     - Interoperability
     - Quality
     - Performance
     - Reliability
     - Fault tolerance
     - Safety
     - Usability
  - Operating
     - System administration
     - Incident response
     - Continuous Integration/Deployment
     - DevOps
     - Containerization

>   - Process development (best practices, software quality, etc.)
>   - Support

Reviewer 2: ```suggestion
  - Process development (best practices, software quality, etc.)
  - Code review
  - Support
```

>     - End-user support
>     - Maintenance of support systems

Reviewer 2: ```suggestion
    - End-user support
    - Contributor support
    - Maintenance of support systems
```
Perhaps a skill related to helping new contributors learn how to make contributions to a software project?

  - Documentation
    - Developer documentation
    - End-user documentation
    - Reports (stats, progress reports, etc.)

**Data management systems/information storage systems**
  - Database design and models
      - Relational
      - NoSQL
      - Document Repositories
  - Data structures
  - Query languages
  - Database administration

**Data**
  - Data visualization
  - Data management
  - Data analysis

**People-related**
  - People management
  - Technical leadership
  - Software advertising
  - Software feedback gathering
  - Training
     - Teaching
     - Consulting

>      - Mentoring
>   - Funding/Grants

Reviewer 2: ```suggestion
     - Mentoring
     - Pair programming
  - Funding/Grants
```
Not sure...this might fit?

---
layout: page
title: RSE Skills
permalink: /resources/skills/
subtitle: Skillsets used by RSEs
---

The following is a list of skills that an RSE might possess. The US-RSE Education and Training Group maintains and publishes this list in an effort to better define RSE roles and to help direct individuals looking to invest in learning new skills. An RSE typically has experience with a subset of these skills. This list is not intended to define what an RSE should know, but rather as a list that can be used to create a profile or job description by choosing relevant skills. This list is also not comprehensive. If you have additions, please contact the Training and Education Working Group.

## Software Development
  - Requirements gathering/analysis/formulation
  - Technology evaluation
  - Packaging/Releasing
  - Programming/implementation
    - Software Architecture
    - Cross-platform development
    - Maintenance
        - Bug fixing
        - Version upgrades
  - Testing

>   - Distributed systems

Reviewer 4: Add "Peer code review"

---
layout: page

> title: Skill Sets used by RSEs

Reviewer 4: ```suggestion
title: Skills Used by RSEs
```

permalink: /resources/training/skills/

> subtitle: Skill Sets used by RSEs

Reviewer 4: ```suggestion
subtitle: Skills Used by RSEs
```

---

> The following is a list of skills that an RSE might possess. The US-RSE Education and Training Group maintains and publishes this list in an effort to better define RSE roles and to help direct individuals looking to invest in learning new skills. An RSE typically has experience with a subset of these skills. This list is not intended to define what an RSE should know, but rather as a list that can be used to create a profile or job description by choosing relevant skills. This list is also not comprehensive. If you have additions, please contact the [US-RSE Training and Education Working Group](({{ site.baseurl }}/about/working-groups/)).

Reviewer: ```suggestion
The following is a list of skills that an RSE might possess. The US-RSE Education and Training Group maintains and publishes this list in an effort to provide resources to better define RSE roles and to help direct individuals looking to invest in learning new skills. An RSE typically has experience with a subset of these skills. This list can be used to create a profile or job description, help individual identify topics they wish to learn, and (eventually) provide links to relevant training resources. This list is a work in progress. If you have additions or suggestions for topics or existing training resources, please either create a pull request or contact the [US-RSE Training and Education Working Group](({{ site.baseurl }}/about/working-groups/)).
```

Reviewer 4: This pull request is the result of the Education & Training working group meeting today. It is ready to be merged.

> The list was originally developed as part of the 2nd International RSE Leaders Workshop held in  2020. It is based off of the analysis of several job postings as well as crowed sourced.

Reviewer 2: ```suggestion
The list was originally developed as part of the 2nd International RSE Leaders Workshop held in 2020. It is based off of the analysis of several job postings as well as crowd sourced.
```
